### PR TITLE
ci: use ubuntu 20.04 for nightly builds on linux

### PR DESCRIPTION
**Pull request by mukilan**
Ubuntu 22.04 has a newer glibc (2.34) which means builds from there won't run on systems with older glibc, most notably the wpt.fyi taskcluster runners which use 20.04 as the docker base image.

This is a temporary workaround until wpt upgrades to 22.04

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes do not require tests because they only impact nighty CI job>

### CODE REVIEW

**Review by mrobinson — APPROVED** [review]

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -61,7 +61,7 @@ env:
 jobs:
   build:
     name: Linux Build
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-${{ inputs.upload && '20.04' || '22.04' }}
```
[the remark this answers is not in the document]
Comment (mukilan) [reply .github/workflows/linux.yml:67]: done in 0e40516

**Comment by sagudev** [comment]
This might be also worth keeping after wpt.fyi starts using 22.04 so we keep nightly working on more computers for as long as possible.

**Comment by sagudev** [comment]
It fails: https://github.com/servo/servo/actions/runs/7014572603/job/19082505564#step:10:6151

**Comment by sagudev** [comment]
Set LIBCLANG_PATH env step should be skipped in ubuntu 22.04 (due to older version of llvm that is already correctly detected IIRC).

**Comment by mukilan** [comment]
Yep, noticed the same and have raised [#30792](https://github.com/servo/servo/pull/30792)

**Pull request merged**